src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -401,6 +407,61 @@ private void EmitSequencePointStatement(BoundSequencePointWithSpan node)
             }
         }
 
+        private void EmitSaveSequencePoint(BoundSaveSequencePoint statement)
+        {
+            if (!_emitPdbSequencePoints)
+                return;
+
+            ArrayBuilder<RawSequencePoint> sequencePoints = _builder.SeqPointsOpt;
+            if (sequencePoints is null)
+                return;
+
+            for (int i = sequencePoints.Count - 1; i >= 0; i--)
+            {
+                var span = sequencePoints[i].Span;
+                if (span == RawSequencePoint.HiddenSequencePointSpan)
+                    continue;
+
+                // Found the previous non-hidden sequence point.  Save it.
+                _savedSequencePoints ??= PooledDictionary<object, TextSpan>.GetInstance();
+                _savedSequencePoints.Add(statement.Identifier, span);
+                return;
+            }
+
+            // failed to find a previous sequence point.  This can occur for example if
+            // a switch expression is the entire body of an expression-bodied method.
+            // In this case we do not need to save the sequence point because restoring
+            // it would occur after any generated code.
+            return;
+        }
+
+        private void EmitRestorePreviousSequencePoint(BoundRestorePreviousSequencePoint node)
+        {
+            Debug.Assert(node.Syntax is { });
+            if (!_emitPdbSequencePoints)
+                return;
+
+            if (_savedSequencePoints is null || !_savedSequencePoints.TryGetValue(node.Identifier, out var span))
+                return;
+
+            EmitStepThroughSequencePoint(node.Syntax.SyntaxTree, span);
+        }
+
+        private void EmitStepThroughSequencePoint(BoundStepThroughSequencePoint node)
+        {
+            EmitStepThroughSequencePoint(node.Syntax.SyntaxTree, node.Span);
+        }
+
+        private void EmitStepThroughSequencePoint(SyntaxTree syntaxTree, TextSpan span)
+        {
+            var label = new object();
+            _builder.EmitBranch(ILOpCode.Br, label);
+            EmitSequencePoint(syntaxTree, span);
+            _builder.EmitOpCode(ILOpCode.Nop);
```

Comment:
>_builder [](start = 12, length = 8)

This doesn't quite work: the emitter won't emit unreachable code.  I may have to place an (unused) label before it to cause it to be emitted.

Reply:
Bu that doesn't work either :(

---
In reply to: [397269804](https://github.com/dotnet/roslyn/pull/42711#discussion_r397269804) [](ancestors = 397269804)

Reply:
This works (Iteration 6)

---
In reply to: [397345228](https://github.com/dotnet/roslyn/pull/42711#discussion_r397345228) [](ancestors = 397345228,397269804)